highfidelity/hifi
# Various shadow bug fixes

This PR fixes bugs:
- [#17108](https://highfidelity.manuscript.com/f/cases/17108/Shadows-don-t-render-if-shadow-frustum-starts-inside-opaque-object-s-bounding-box)
- [#16723](https://highfidelity.manuscript.com/f/cases/16723/Rendering-Shadow-Grazing-test-image-doesn-t-match-expected-test-image)
- [#17106](https://highfidelity.manuscript.com/f/cases/17106/Grazing-angle-shadows)

It partially fixes bug [#17286](https://highfidelity.fogbugz.com/f/cases/17286/Shadow-resolution-bug-and-other-shadow-issues#BugEvent.113869) (only the faint rectangular outline bug, not the loss of resolution).

It could also possibly fix bug [#16198](https://highfidelity.manuscript.com/f/cases/16198/Shadow-rendering-issue-in-Maker-domain) but I can't verify this as the Maker domain's protocol isn't compatible with master.

The PR also fixes the Luci visualisation of the depth buffer and shadow cascade indices.

# TEST PLAN
1. Run all _tests/engine/render/shadows_ tests from this update in [hifi_tests PR #248](https://github.com/highfidelity/hifi_tests/pull/248)
2. Also: go to a domain where shadows are visible (Home for instance), run the _developer\utilities\workload\workload.js_ script as described in bug report #17108, click the Workload button on the task bar 
- in the newly appeared app, check the "Show views" checkbox: shadows should remain the same and still be visible
- in the newly appeared app, check the "Show proxies" checkbox: shadows should remain the same and still be visible

3. Run all _tests/content/overlay_ tests from master as the fix for Luci may have some incidence on the overlays.
